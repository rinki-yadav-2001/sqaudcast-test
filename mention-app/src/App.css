.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}
.input {
  font-size: 16px;
  color: #282c34;
  width: 70%;
  height: 40px;
  border: #61dafb 2px solid;
}
.mentionContainer {
  margin-top: 5px;
  width: 69%;
  padding: 40px 10px;
  columns: rgb(30, 3, 3);
  font-size: 18px;
  border: #61dafb 1px solid;
  box-shadow: #282c34;
}
.option {
  padding: 10px;
  columns: rgb(30, 3, 3);
  font-size: 18px;
}
@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
