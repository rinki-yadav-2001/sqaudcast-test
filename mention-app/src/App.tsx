import React, { useState } from "react";
import MentionInput from "./components/MentionInput";
import MentionSuggestions from "./components/MentionComponent";
import data from "./data.json";
import "./App.css";

interface MentionOption {
  id: number;
  name: string;
}
const App: React.FC = () => {
  const [inputValue, setInputValue] = useState<string>("");
  const [mentionOptions, setMentionOptions] = useState<MentionOption[]>([]);
  const [showOptions, setShowOptions] = useState<boolean>(false);

  const handleInputChange = (value: string) => {
    setInputValue(value);
    if (value.includes("@")) {
      setShowOptions(true);
      const filteredOptions = data.filter((option) =>
        option.name.toLowerCase().includes(value.toLowerCase().replace("@", ""))
      );

      setMentionOptions(filteredOptions);
    } else {
      setShowOptions(false);
    }
  };

  const handleOptionSelect = (option: MentionOption) => {
    const updatedText = inputValue.replace(/@[\w]+/, `@${option.name}`);
    setInputValue(updatedText);
    setShowOptions(false);
  };

  return (
    <div className="App">
      <MentionInput value={inputValue} onChange={handleInputChange} />
      {showOptions && (
        <MentionSuggestions
          options={mentionOptions}
          onSelect={handleOptionSelect}
        />
      )}
    </div>
  );
};

export default App;
