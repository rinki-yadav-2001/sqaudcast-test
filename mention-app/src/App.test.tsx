import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from "./App";

describe("App component", () => {
  test("renders input field", () => {
    const { getByPlaceholderText } = render(<App />);

    const inputElement = getByPlaceholderText("Mention...");
    expect(inputElement).toBeInTheDocument();
  });

  test("updates input value on change", () => {
    const { getByPlaceholderText } = render(<App />);
    const inputElement = getByPlaceholderText("Mention...");
    fireEvent.change(inputElement, { target: { value: "@Luke" } });

    expect(inputElement).toHaveValue("@Luke");
  });

  test("selects mention option", () => {
    const { getByPlaceholderText, asFragment, getByText } = render(<App />);
    const inputElement = getByPlaceholderText("Mention...");
    fireEvent.change(inputElement, { target: { value: "@Luke" } });
    const mentionOption = getByText("Luke Skywalker");
    expect(mentionOption).toBeInTheDocument();

    fireEvent.click(mentionOption);

    const componentSnapshot = asFragment();
    expect(componentSnapshot).toMatchSnapshot();
    expect(inputElement).toHaveValue("@Luke Skywalker");
  });
});
