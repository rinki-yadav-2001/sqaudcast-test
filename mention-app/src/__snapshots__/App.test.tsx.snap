// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component selects mention option 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <input
      class="input"
      placeholder="Mention..."
      type="text"
      value="@Luke Skywalker"
    />
  </div>
</DocumentFragment>
`;
